packages/paste-core/components/table/src/types.ts

```diff
@@ -20,24 +20,61 @@ export interface TableContextProps {
 
 export interface TableProps extends HTMLPasteProps<'table'> {
   children: NonNullable<React.ReactNode>;
+  /**
+   * Overrides the default element name ('TABLE') to apply unique styles with the Customization Provider.
+   *
+   * @default TABLE
+   * @type {BoxProps['element']}
+   * @memberof TableProps
+   */
   element?: BoxProps['element'];
+  /**
+   * Sets a background color for even rows within the table.
+   *
+   * @type {boolean}
+   * @memberof TableProps
+   */
   striped?: boolean;
+  /**
+   * Sets the `table-layout` style of the Table. Defaults to `auto`.
+   *
+   * @type {TableLayoutOptions}
+   * @memberof TableProps
+   */
   tableLayout?: TableLayoutOptions;
+  /**
+   * Sets the `border` style of the Table. Defaults to `default`.
+   *
+   * @type {TableVariantOptions}
+   * @memberof TableProps
+   */
   variant?: TableVariantOptions;
   /**
-   * Controls a tables ability to be shown on smaller viewports
+   * Sets the table to scroll horizontally on small screens.
    */
   scrollHorizontally?: boolean;
   /**
-   * Controls table cell content line wrapping
+   * Sets the table cells to not line wrap.
    */
   noWrap?: boolean;
-  /** Displays table and data grid actionable mode */
+  /**
+   * Sets the table to visually display the actionable state of an interactive table. Mainly used for Data Grid.
+   *
+   * @type {boolean}
+   * @memberof TableProps
+   */
   isActionable?: boolean;
 }
 
 export interface THeadProps extends HTMLPasteProps<'thead'> {
   children: NonNullable<React.ReactNode>;
+  /**
+   * Overrides the default element name ('THEAD') to apply unique styles with the Customization Provider.
+   *
+   * @default THEAD
+   * @type {BoxProps['element']}
+   * @memberof THeadProps
+   */
   element?: BoxProps['element'];
```

Comment:
suggestion: as we go through these changes since we'll be touching all component types, can we change all the `element` types to be how this one is typed instead of Pick<BoxProps, 'element'>? That way they'll all have default values which I think is most important and we can just copy/paste the description

Reply:
meaning Button would change in this PR